velox/functions/prestosql/tests/CustomTypeCastTest.cpp

```diff
@@ -0,0 +1,215 @@
+    std::shared_ptr<const core::ITypedExpr> inputField =
+        std::make_shared<const core::FieldAccessTypedExpr>(fromType, "c0");
+    std::shared_ptr<const core::ITypedExpr> typedExpr =
+        std::make_shared<const core::CastTypedExpr>(
+            toType,
+            std::vector<std::shared_ptr<const core::ITypedExpr>>{inputField},
+            tryCast);
+
+    auto result = evaluate<SimpleVector<EvalType<TTo>>>(typedExpr, input);
+
+    assertEqualVectors(expected, result);
+  }
+
+  template <typename TFrom, typename TTo>
+  void testCustomCast(
+      const TypePtr& fromType,
+      const TypePtr& toType,
+      std::vector<std::optional<TFrom>> input,
+      std::vector<std::optional<TTo>> expected,
+      bool tryCast = false) {
+    evaluateCustomCast<TTo>(
+        fromType,
+        toType,
+        makeRowVector({makeNullableFlatVector<TFrom>(input)}),
+        makeNullableFlatVector<TTo>(expected),
+        tryCast);
+  }
+
+  template <typename TFrom, typename TTo>
+  void testCustomCastConstEncoding(
+      const TypePtr& fromType,
+      const TypePtr& toType,
+      std::vector<std::optional<TFrom>> input,
+      std::vector<std::optional<TTo>> expected,
+      bool tryCast = false) {
+    auto inputVector = makeNullableFlatVector<TFrom>(input);
+    auto constInput = BaseVector::wrapInConstant(5, 0, inputVector);
+
+    auto expectedVector = makeNullableFlatVector<TTo>(expected);
+    auto constExpected = BaseVector::wrapInConstant(5, 0, expectedVector);
+
+    evaluateCustomCast<TTo>(
+        fromType, toType, makeRowVector({constInput}), constExpected, tryCast);
+  }
+
+  template <typename TFrom, typename TTo>
+  void testCustomCastDictEncoding(
+      const TypePtr& fromType,
+      const TypePtr& toType,
+      std::vector<std::optional<TFrom>> input,
+      std::vector<std::optional<TTo>> expected,
+      bool tryCast = false) {
+    std::shared_ptr<const core::ITypedExpr> inputField =
+        std::make_shared<const core::FieldAccessTypedExpr>(fromType, "c0");
+    std::shared_ptr<const core::ITypedExpr> callExpr =
+        std::make_shared<const core::CallTypedExpr>(
+            fromType,
+            std::vector<std::shared_ptr<const core::ITypedExpr>>{inputField},
+            "testing_dictionary");
+    std::shared_ptr<const core::ITypedExpr> typedExpr =
+        std::make_shared<const core::CastTypedExpr>(
+            toType,
+            std::vector<std::shared_ptr<const core::ITypedExpr>>{callExpr},
+            tryCast);
+
+    auto inputVector = makeRowVector({makeNullableFlatVector<TFrom>(input)});
+
+    auto expectedVector = makeNullableFlatVector<TTo>(expected);
+    auto indices = ::makeIndicesInReverse(inputVector->size(), pool());
+
+    auto result = evaluate<SimpleVector<EvalType<TTo>>>(typedExpr, inputVector);
+    assertEqualVectors(
+        wrapInDictionary(indices, inputVector->size(), expectedVector), result);
+  }
+};
+
+TEST_F(CastExprTest, testJsonCustomCastInteger) {
```

Comment:
TEST_F(JsonCastTest, fromBigint) for clarity

Reply:
Renamed. Thanks!